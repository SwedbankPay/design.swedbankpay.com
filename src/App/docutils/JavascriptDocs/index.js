import React from "react";
import { Link } from "react-router-dom";

import CodeTags from "@components/CodeTags";

import DocElement from "@docutils/JavascriptDocElement";

const OpenDocs = ({ componentName }) => (
    <>
        <DocElement
            code={<CodeTags type="secondary" code={`dg.${componentName}.open(<${componentName}-id>)`} params={[`<${componentName}-id>`]} />}
            description={
                <>
                    Opens the given {componentName.toLowerCase()}
                </>
            }
        />
    </>
);

const CloseDocs = ({ componentName }) => (
    <>
        <DocElement
            code={<CodeTags type="secondary" code={`dg.${componentName}.close(<${componentName}-id>)`} params={[`<${componentName}-id>`]} />}
            description={
                <>
                    Closes the given {componentName}
                </>
            }
        />
    </>
);

const JavascriptDocs = ({ componentName, open, close, others }) => (
    <>
        <ul className="item-list">
            <DocElement
                code={<CodeTags type="secondary" code={`dg.${componentName}.init()`} />}
                description={
                    <>
                        <Link to="/docs/getting-started/javascript">Initializes</Link> {componentName}
                    </>
                }
            />
            {open ? <OpenDocs componentName={componentName} /> : null}
            {close ? <CloseDocs componentName={componentName} /> : null}
            {others && others.map((DocComponent, i) => <DocComponent key={i} componentName={componentName} />)}
        </ul>
    </>
);

export default JavascriptDocs;

/* For testing */
export { OpenDocs, CloseDocs };
