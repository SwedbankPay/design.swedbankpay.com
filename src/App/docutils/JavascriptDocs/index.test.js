import React from "react";
// import { Link } from "react-router-dom";
import { shallow, mount } from "enzyme";

import JavascriptDocs, { OpenDocs, CloseDocs } from "./index";

jest.mock("react-router-dom");

describe("Utilities: JavascriptDocs", () => {
    const componentName = "test";

    it("is defined", () => {
        expect(JavascriptDocs).toBeDefined();
    });

    it("renders only init method", () => {
        const wrapper = shallow(<JavascriptDocs componentName="test" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<OpenDocs componentName={componentName} />)).toBeFalsy();
        expect(wrapper.contains(<CloseDocs componentName={componentName} />)).toBeFalsy();
    });

    it("renders init and open methods", () => {
        const wrapper = mount(
            <JavascriptDocs componentName={componentName} open />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<OpenDocs componentName={componentName} />)).toBeTruthy();
        expect(wrapper.contains(<CloseDocs componentName={componentName} />)).toBeFalsy();
    });

    it("renders init and close methods", () => {
        const wrapper = mount(
            <JavascriptDocs componentName={componentName} close />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<OpenDocs componentName={componentName} />)).toBeFalsy();
        expect(wrapper.contains(<CloseDocs componentName={componentName} />)).toBeTruthy();
    });

    it("renders init, open and close methods", () => {
        const wrapper = mount(
            <JavascriptDocs componentName={componentName} open close />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<OpenDocs componentName={componentName} />)).toBeTruthy();
        expect(wrapper.contains(<CloseDocs componentName={componentName} />)).toBeTruthy();
    });

    describe("OpenDocs", () => {
        it("is defined", () => {
            expect(OpenDocs).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<OpenDocs componentName={componentName} />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("CloseDocs", () => {
        it("is defined", () => {
            expect(CloseDocs).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<CloseDocs componentName={componentName} />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});

