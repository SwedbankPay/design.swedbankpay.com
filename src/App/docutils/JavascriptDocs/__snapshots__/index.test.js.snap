// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utilities: JavascriptDocs CloseDocs renders 1`] = `
<Fragment>
  <JavascriptDocElement
    code={
      <CodeTags
        code="dg.test.close(<test-id>)"
        params={
          Array [
            "<test-id>",
          ]
        }
        type="secondary"
      />
    }
    description={
      <React.Fragment>
        Closes the given 
        test
      </React.Fragment>
    }
  />
</Fragment>
`;

exports[`Utilities: JavascriptDocs OpenDocs renders 1`] = `
<Fragment>
  <JavascriptDocElement
    code={
      <CodeTags
        code="dg.test.open(<test-id>)"
        params={
          Array [
            "<test-id>",
          ]
        }
        type="secondary"
      />
    }
    description={
      <React.Fragment>
        Opens the given 
        test
      </React.Fragment>
    }
  />
</Fragment>
`;

exports[`Utilities: JavascriptDocs renders init and close methods 1`] = `
<JavascriptDocs
  close={true}
  componentName="test"
>
  <ul
    className="item-list"
  >
    <JavascriptDocElement
      code={
        <CodeTags
          code="dg.test.init()"
          type="secondary"
        />
      }
      description={
        <React.Fragment>
          <Link
            to="/getting-started/javascript"
          >
            Initializes
          </Link>
           
          test
        </React.Fragment>
      }
    >
      <li>
        <div>
          <h3
            scope="row"
          >
            <CodeTags
              code="dg.test.init()"
              type="secondary"
            >
              <code
                className="code-tags code-tags-secondary"
              >
                dg.test.init()
              </code>
            </CodeTags>
          </h3>
          <p
            className="javascript-doc-element-description"
          >
            <Link
              to="/getting-started/javascript"
            />
             
            test
          </p>
        </div>
      </li>
    </JavascriptDocElement>
    <CloseDocs
      componentName="test"
    >
      <JavascriptDocElement
        code={
          <CodeTags
            code="dg.test.close(<test-id>)"
            params={
              Array [
                "<test-id>",
              ]
            }
            type="secondary"
          />
        }
        description={
          <React.Fragment>
            Closes the given 
            test
          </React.Fragment>
        }
      >
        <li>
          <div>
            <h3
              scope="row"
            >
              <CodeTags
                code="dg.test.close(<test-id>)"
                params={
                  Array [
                    "<test-id>",
                  ]
                }
                type="secondary"
              >
                <code
                  className="code-tags code-tags-secondary"
                >
                  dg.test.close(&lt;test-id&gt;)
                </code>
              </CodeTags>
            </h3>
            <p
              className="javascript-doc-element-description"
            >
              Closes the given 
              test
            </p>
          </div>
        </li>
      </JavascriptDocElement>
    </CloseDocs>
  </ul>
</JavascriptDocs>
`;

exports[`Utilities: JavascriptDocs renders init and open methods 1`] = `
<JavascriptDocs
  componentName="test"
  open={true}
>
  <ul
    className="item-list"
  >
    <JavascriptDocElement
      code={
        <CodeTags
          code="dg.test.init()"
          type="secondary"
        />
      }
      description={
        <React.Fragment>
          <Link
            to="/getting-started/javascript"
          >
            Initializes
          </Link>
           
          test
        </React.Fragment>
      }
    >
      <li>
        <div>
          <h3
            scope="row"
          >
            <CodeTags
              code="dg.test.init()"
              type="secondary"
            >
              <code
                className="code-tags code-tags-secondary"
              >
                dg.test.init()
              </code>
            </CodeTags>
          </h3>
          <p
            className="javascript-doc-element-description"
          >
            <Link
              to="/getting-started/javascript"
            />
             
            test
          </p>
        </div>
      </li>
    </JavascriptDocElement>
    <OpenDocs
      componentName="test"
    >
      <JavascriptDocElement
        code={
          <CodeTags
            code="dg.test.open(<test-id>)"
            params={
              Array [
                "<test-id>",
              ]
            }
            type="secondary"
          />
        }
        description={
          <React.Fragment>
            Opens the given 
            test
          </React.Fragment>
        }
      >
        <li>
          <div>
            <h3
              scope="row"
            >
              <CodeTags
                code="dg.test.open(<test-id>)"
                params={
                  Array [
                    "<test-id>",
                  ]
                }
                type="secondary"
              >
                <code
                  className="code-tags code-tags-secondary"
                >
                  dg.test.open(&lt;test-id&gt;)
                </code>
              </CodeTags>
            </h3>
            <p
              className="javascript-doc-element-description"
            >
              Opens the given 
              test
            </p>
          </div>
        </li>
      </JavascriptDocElement>
    </OpenDocs>
  </ul>
</JavascriptDocs>
`;

exports[`Utilities: JavascriptDocs renders init, open and close methods 1`] = `
<JavascriptDocs
  close={true}
  componentName="test"
  open={true}
>
  <ul
    className="item-list"
  >
    <JavascriptDocElement
      code={
        <CodeTags
          code="dg.test.init()"
          type="secondary"
        />
      }
      description={
        <React.Fragment>
          <Link
            to="/getting-started/javascript"
          >
            Initializes
          </Link>
           
          test
        </React.Fragment>
      }
    >
      <li>
        <div>
          <h3
            scope="row"
          >
            <CodeTags
              code="dg.test.init()"
              type="secondary"
            >
              <code
                className="code-tags code-tags-secondary"
              >
                dg.test.init()
              </code>
            </CodeTags>
          </h3>
          <p
            className="javascript-doc-element-description"
          >
            <Link
              to="/getting-started/javascript"
            />
             
            test
          </p>
        </div>
      </li>
    </JavascriptDocElement>
    <OpenDocs
      componentName="test"
    >
      <JavascriptDocElement
        code={
          <CodeTags
            code="dg.test.open(<test-id>)"
            params={
              Array [
                "<test-id>",
              ]
            }
            type="secondary"
          />
        }
        description={
          <React.Fragment>
            Opens the given 
            test
          </React.Fragment>
        }
      >
        <li>
          <div>
            <h3
              scope="row"
            >
              <CodeTags
                code="dg.test.open(<test-id>)"
                params={
                  Array [
                    "<test-id>",
                  ]
                }
                type="secondary"
              >
                <code
                  className="code-tags code-tags-secondary"
                >
                  dg.test.open(&lt;test-id&gt;)
                </code>
              </CodeTags>
            </h3>
            <p
              className="javascript-doc-element-description"
            >
              Opens the given 
              test
            </p>
          </div>
        </li>
      </JavascriptDocElement>
    </OpenDocs>
    <CloseDocs
      componentName="test"
    >
      <JavascriptDocElement
        code={
          <CodeTags
            code="dg.test.close(<test-id>)"
            params={
              Array [
                "<test-id>",
              ]
            }
            type="secondary"
          />
        }
        description={
          <React.Fragment>
            Closes the given 
            test
          </React.Fragment>
        }
      >
        <li>
          <div>
            <h3
              scope="row"
            >
              <CodeTags
                code="dg.test.close(<test-id>)"
                params={
                  Array [
                    "<test-id>",
                  ]
                }
                type="secondary"
              >
                <code
                  className="code-tags code-tags-secondary"
                >
                  dg.test.close(&lt;test-id&gt;)
                </code>
              </CodeTags>
            </h3>
            <p
              className="javascript-doc-element-description"
            >
              Closes the given 
              test
            </p>
          </div>
        </li>
      </JavascriptDocElement>
    </CloseDocs>
  </ul>
</JavascriptDocs>
`;

exports[`Utilities: JavascriptDocs renders only init method 1`] = `
<Fragment>
  <ul
    className="item-list"
  >
    <JavascriptDocElement
      code={
        <CodeTags
          code="dg.test.init()"
          type="secondary"
        />
      }
      description={
        <React.Fragment>
          <Link
            to="/getting-started/javascript"
          >
            Initializes
          </Link>
           
          test
        </React.Fragment>
      }
    />
  </ul>
</Fragment>
`;
